### PR TITLE
Ability to add and consume meta info for Endpoints

### PR DESCRIPTION
**Pull request by shrey-garg**

### CODE REVIEW

**Comment by pmohankumar** [comment]
As discussed offline, lets add additional properties under a key in API definition (instead of polluting global namespace which poseidon alone should have control). Key could be "properties" or "additionalProperties".

Lets also demonstrate a use case under sample application. Add a key value in API file and read it through context and log it in a DataSource & Filter.

File: api/src/main/java/com/flipkart/poseidon/api/APILegoSet.java [hunk 1]
```diff
@@ -102,8 +99,10 @@ public Buildable getBuildable(Request request) throws LegoSetException, ElementN
             EndpointPOJO pojo = ((APIBuildable) buildable).getPojo();
             RequestContext.set(URI, pojo.getUrl());
             RequestContext.set(ENDPOINT_NAME, pojo.getName());
+            RequestContext.set(API_ANNOTATIONS, pojo.getProperties());
             RequestContext.set(ENDPOINT_METHOD, pojo.getHttpMethod());
             ServiceContext.set(ENDPOINT_NAME, pojo.getName());
+            ServiceContext.set(API_ANNOTATIONS, pojo.getProperties());
```
Comment (pmohankumar) [inline api/src/main/java/com/flipkart/poseidon/api/APILegoSet.java]: Do we really need it in service context? Service context is to be consumed by poseidon framework classes like AbstractServiceClient etc where request context isn't available. We need to be frugal on thread locals.

Comment (shrey-garg) [inline api/src/main/java/com/flipkart/poseidon/api/APILegoSet.java]: Agreed, RequestContext is available everywhere a user might want to use these.

File: api/src/main/java/com/flipkart/poseidon/pojos/EndpointPOJO.java [hunk 2]
```diff
@@ -83,4 +87,12 @@ public ParamsPOJO getParams() {
     public Object getResponse() {
         return response;
     }
+
+    public Map<String, Object> getProperties() {
+        return properties;
+    }
+
+    public void addAdditionalProperties(String name, Object value) {
```
Comment (pmohankumar) [inline api/src/main/java/com/flipkart/poseidon/pojos/EndpointPOJO.java]: This isn't required?

**Pull request merged**